docroot/modules/custom/foia_file/drush/foia_file.drush.inc

```diff
@@ -0,0 +1,158 @@
+<?php
+
+/**
+ * @file
+ * FOIA Api module drush command.
+ */
+
+use Drupal\file\Entity\File;
+
+/**
+ * Implements hook_drush_command().
+ */
+function foia_file_drush_command() {
+  $items = [];
+
+  $items['file-entity-update'] = [
+    'description' => 'Updates file entities virus scan statuses.',
+    'core' => ['8+'],
+    'arguments' => [
+      'scan_result' => 'The result of the file scan.',
+    ],
+    'aliases' => ['feu'],
+  ];
+
+  return $items;
+}
+
+/**
+ * Implements hook_drush_help().
+ */
+function foia_file_drush_help($command) {
+  switch ($command) {
+    case 'drush:file-entity-update':
+      return dt('Update virus scan statuses for file entites based upon output from virus scanner.');
+  }
+}
+
+/**
+ * Implements drush_hook_COMMAND().
+ */
+function drush_foia_file_file_entity_update($scanOutput) {
+  $startTime = microtime(TRUE);
+
+  // Get webform directory on the server.
+  $fileDir = explode('/webform/', $scanOutput, 2);
+  $webformDir = "{$fileDir[0]}/webform/";
+  // Temporarily delete similar paths of the webform dir during processing.
+  $trimmedFiles = str_replace($webformDir, '', $scanOutput);
+
+  // Split string into lines. @todo change to preg split \R?
+  $scans = explode("\n", $trimmedFiles);
+
+  // Set constants.
+  $cleanScanStatuses = ['OK', 'Empty file'];
+  $doNotScanEntityStatuses = ['clean', 'virus'];
+
+  foreach ($scans as $scan) {
+
+    // Validate the scan output status.
+    preg_match('/ (OK|Empty file|FOUND|Removed\.)$/', $scan, $scanStatusWithLeadingSpace);
+    // Do not process scan status "FOUND".
+    // These will be processed under the duplicate "Remove." scan status.
+    if ($scanStatusWithLeadingSpace[0] === ' FOUND') {
+      continue;
+    }
+    if ($scanStatusWithLeadingSpace) {
+      // Trim the leading space.
+      $scanStatus = substr($scanStatusWithLeadingSpace[0], 1);
+    }
+    else {
+      \Drupal::logger('foia_file')->warning(
+        "An unexpected scan status was detected in the virus scan output."
+      );
+      continue;
+    }
+
+    $relativeFileName = preg_replace('/: (OK|Empty file|Removed\.)$/', '', $scan);
+    $absoluteFileName = $webformDir . $relativeFileName;
+    $pathParts = pathinfo($absoluteFileName);
+
+    // Log any discrepancies between filesystem and entities.
+    if (in_array($scanStatus, $cleanScanStatuses) && !file_exists($absoluteFileName)) {
+      \Drupal::logger('foia_file')->warning(
+        "The file @absoluteFileName was not detected on the filesystem but was marked as clean by the virus scanner.",
```

Comment:
I just saw this comment. This seems to happen in the test environments when file entities do not necessarily match the filesystem. In real life, this might happen if a file is deleted or corrupted on the filesystem. Ideally, a notification similar to that of when a file is intentionally deleted should be sent to the agency component so they can know the file is missing.